packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -1072,6 +1117,114 @@ class ScrollAction extends Action<ScrollIntent> {
   }
 }
 
+/// Provides a fallback [Scrollable] for [ScrollAction]s.
+///
+/// The PrimaryScrollShortcut can only provide one Scrollable to ScrollActions.
+///
+// TODO(Piinks): More docs and examples
+///
+/// See also:
+///
+///  * [Scrollable.primaryScrollShortcut], a flag that, when true, will register
+///    the associated Scrollable widget as a fallback for [ScrollAction]s.
+class PrimaryScrollShortcut extends StatefulWidget {
+  /// Create a widget for managing a fallback [ScrollAction].
+  const PrimaryScrollShortcut({
+    Key? key,
+    required this.child
+  }) : super(key: key);
+
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  /// The state from the closest instance of this class that encloses the given
+  /// context.
+  ///
+  /// If there is no [PrimaryScrollShortcut] in scope, then this will assert in
+  /// debug mode, and throw an exception in release mode.
+  ///
+  /// See also:
+  ///
+  ///  * [maybeOf], which is a similar function but will return null instead of
+  ///    throwing if there is no [PrimaryScrollShortcut] ancestor.
+  static _PrimaryScrollShortcutState of(BuildContext context) {
+    assert(context != null);
+    final _PrimaryScrollShortcutScope? scope = context.dependOnInheritedWidgetOfExactType<_PrimaryScrollShortcutScope>();
```

Comment:
Since you don't really care about creating a dependency here, you could use `context.getElementForInheritedWidgetOfExactType<_PrimaryScrollShortcutScope>().widget`, which might be just a little faster, since it doesn't have to add it to the dependencies.